app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt

```diff
@@ -55,37 +55,43 @@ class EarnViewModel(val taskRepository: TasksRepository, val wallet: Wallet,
         taskRepository.taskState = TaskState.IN_PROGRESS
         val task = taskRepository.task
         val bEvent = Events.Business.EarningTaskStarted(
-            task?.provider?.name,
-            task?.minToComplete,
-            task?.kinReward,
-            task?.tagsString(),
-            task?.id,
-            task?.title,
-            task?.type)
+                task?.provider?.name,
+                task?.minToComplete,
+                task?.kinReward,
+                task?.tagsString(),
+                task?.id,
+                task?.title,
+                task?.type)
         analytics.logEvent(bEvent)
 
         val aEvent = Events.Analytics.ClickStartButtonOnTaskPage(
-            isTaskStarted.get(),
-            task?.provider?.name,
-            task?.minToComplete,
-            task?.kinReward,
-            task?.tagsString(),
-            task?.id,
-            task?.title,
-            task?.type)
+                isTaskStarted.get(),
+                task?.provider?.name,
+                task?.minToComplete,
+                task?.kinReward,
+                task?.tagsString(),
+                task?.id,
+                task?.title,
+                task?.type)
         analytics.logEvent(aEvent)
         navigator.navigateTo(Navigator.Destination.TASK)
     }
 
     private fun refresh() {
-        taskRepository.task?.let {
-            isQuiz.set(it.isQuiz())
-            authorName.set(it.provider?.name)
-            authorImageUrl.set(it.provider?.imageUrl)
-            title.set(it.title)
-            description.set(it.description)
-            kinReward.set(it.kinReward?.toString())
-            minToComplete.set(convertMinToCompleteToString(it.minToComplete))
+        taskRepository.task?.let { task ->
+            isQuiz.set(task.isQuiz())
+            authorName.set(task.provider?.name)
+            authorImageUrl.set(task.provider?.imageUrl)
+            title.set(task.title)
+            description.set(task.description)
+            kinReward.set(if (task.isQuiz()) {
+                var totalReward = 0
+                (task.questions?.forEach { question ->
+                    totalReward += (question.quiz_data?.reward ?: 0)
+                })
+                totalReward.toString()
+            } else task.kinReward?.toString())
```

Comment:
would be nice to add a test in EarnViewModel test that checks the value of kinReward with different types of tasks